statsmodels/distributions/joint_distribution.py

```diff
@@ -0,0 +1,435 @@
+        points = np.vstack([xx.ravel(), yy.ravel()]).T
+
+        data = self.pdf(points).T.reshape(xx.shape)
+        min_ = np.percentile(data, 5)
+        max_ = np.percentile(data, 95)
+
+        fig, ax = utils.create_mpl_ax(ax)
+
+        vticks = np.linspace(min_, max_, num=ticks_nbr)
+        range_cbar = [min_, max_]
+        cs = ax.contourf(xx, yy, data, vticks,
+                         antialiased=True, vmin=range_cbar[0],
+                         vmax=range_cbar[1])
+
+        ax.set_xlabel("u")
+        ax.set_ylabel("v")
+        ax.set_xlim(0, 1)
+        ax.set_ylim(0, 1)
+        ax.set_aspect('equal')
+        cbar = plt.colorbar(cs, ticks=vticks)
+        cbar.set_label('p')
+
+        return fig
+
+
+class IndependentCopula(Copula):
+    """Independent copula.
+
+    .. math::
+
+        C_\theta(u,v) = uv
+
+    """
+    def __init__(self, d=2):
+        self.d = d
+
+    def random(self, n=1, random_state=None):
+        rng = check_random_state(random_state)
+        x = rng.random((n, self.d))
+        return x
+
+    def pdf(self, x):
+        return np.ones((len(x), 1))
+
+    def cdf(self, x):
+        return np.prod(x, axis=1)
+
+
+class GaussianCopula(Copula):
+    r"""Gaussian copula.
+
+    It is constructed from a multivariate normal distribution over
+    :math:`\mathbb{R}^d` by using the probability integral transform.
+
+    For a given correlation matrix :math:`R \in[-1, 1]^{d \times d}`,
+    the Gaussian copula with parameter matrix :math:`R` can be written
+    as:
+
+    .. math::
+
+        C_R^{\text{Gauss}}(u) = \Phi_R\left(\Phi^{-1}(u_1),\dots,
+        \Phi^{-1}(u_d) \right),
+
+    where :math:`\Phi^{-1}` is the inverse cumulative distribution function
+    of a standard normal and :math:`\Phi_R` is the joint cumulative
+    distribution function of a multivariate normal distribution with mean
+    vector zero and covariance matrix equal to the correlation
+    matrix :math:`R`.
+
+    """
+
+    def __init__(self, cov=None):
+        if cov is None:
+            cov = [[1., 0.], [0., 1.]]
+        self.density = stats.norm()
+        self.mv_density = stats.multivariate_normal(cov=cov)
+
+    def random(self, n=1, random_state=None):
+        x = self.mv_density.rvs(size=n, random_state=random_state)
+        return self.density.cdf(x)
+
+    def pdf(self, x):
+        return self.mv_density.pdf(x)
+
+    def cdf(self, x):
+        return self.mv_density.cdf(x)
+
+
+class StudentCopula(Copula):
+    """Student copula."""
+
+    def __init__(self, df=1, cov=None):
+        if cov is None:
+            cov = [[1., 0.], [0., 1.]]
+        self.density = stats.t(df=df)
+        self.mv_density = stats.multivariate_t(shape=cov, df=df)
+
+    def random(self, n=1, random_state=None):
+        x = self.mv_density.rvs(size=n, random_state=random_state)
+        return self.density.cdf(x)
+
+    def pdf(self, x):
+        return self.mv_density.pdf(x)
```

Comment:
This is not correct, we need the copula pdf and not the pdf of the multivariate t distribution.
I guess we can implement logpdf (loglike) and get pdf through exp(logpdf)